README.md

```diff
@@ -1,27 +1,123 @@
-# Deploy Google Cloud Virtual Machines in a public-only single region with Terraform
+# Deephaven create Image steps
 
-The script will install deephaven application on the virtual machines.
+#1. Login to GCP console and Click to cloud shell where we will run all the commands
 
-network-firewall.tf --> Configure basic firewall for the network
 
-network.tf --> Define network, vpc, subnet, icmp firewall
 
-provider.tf --> Configure Google Cloud provider
+   To set the project property in the core section, run:
+```
+        gcloud config set project YOUR_PROJECT_NAME
+```
 
-terraform.tfvars --> Defining variables
+    To set the zone property in the compute section, run:
+```
+        gcloud config set compute/zone us-central1-a
+```
 
-variables-auth.tf --> Application and authentication variables
+#Setup up project name and zone as environment variable as well
 
-vm-output.tf --> Output of VM
+```
+export PROJECT_ID="YOUR_PROJECT_NAME"
+export ZONE="us-central1-a"
+```
 
-vm.tf --> Create a Ubuntu VM
+#2. Run below command in the cloud shell which will create a GCP vm
 
-# Notes
+```
+gcloud beta compute --project=$PROJECT_ID instances create deephaven --zone=us-central1-a --machine-type=n2-standard-4 --subnet=default --network-tier=PREMIUM --maintenance-policy=MIGRATE --service-account=1088976619727-compute@developer.gserviceaccount.com --scopes=https://www.googleapis.com/auth/devstorage.read_only,https://www.googleapis.com/auth/logging.write,https://www.googleapis.com/auth/monitoring.write,https://www.googleapis.com/auth/servicecontrol,https://www.googleapis.com/auth/service.management.readonly,https://www.googleapis.com/auth/trace.append --tags=deephaven,http-server,https-server --image=centos-7-v20210701 --image-project=centos-cloud --boot-disk-size=30GB --boot-disk-type=pd-balanced --boot-disk-device-name=deephaven --no-shielded-secure-boot --shielded-vtpm --shielded-integrity-monitoring --reservation-affinity=any
+```
 
-# We have used custom image for this VM which has specific required packages installed. Once the VM was ready. We created an image using below command.
 
-gcloud compute images create deephaven    --source-disk=deeephaven-test-instance-1   --source-disk-zone=us-central1-a  --project himantej-development-test
+#3. Login to the vm which we created, use below command to login to vm
 
-Check list of Google Cloud OS images --> https://cloud.google.com/compute/docs/images
+gcloud beta compute ssh --zone "us-central1-a" "deephaven" --project "$PROJECT_ID"
 
-Create the Json file for authentication --> https://cloud.google.com/iam/docs/creating-managing-service-account-keys
+
+#4. Once you login to vm copy and paste below commands and hit enter on the server.
+This will install the required packages and then clone the  deephaven git repo and start the application.
+
+These commands will run for around 20 to 25 min. We have to wait until this is finished, docker build takes time.  
```

Comment:
We can likely speed up the initial image creation process by pulling in the released docker images instead of building them manually.

Reply:
I'm not too worried about the length of time it takes to build the image. Given the earlier PR and code, I assumed that the building was happening at startup, not during the image creation process. But it looks like we are cloning and building during VM creation *and* startup?

IMO, we should only be building (or pulling) the docker images once; arguably, it's best to do it during the image creation process and not on startup.

Reply:
Yes, your opinion is correct. 
We are building the code during VM Image creation (This is not docker image creation) -->This is correct. But if we do not update the VM image regularly then the code will be outdated and keeping multiple images will be very costly for end user.  

The solution to this is ---> 
While creating the VM (Using our custom VM Image) again in the startup script we are pulling the latest deephaven code and build code and start the application. (This will be quick as some packages are already present in the VM image)

We are doing this because we want to keep the code up to date always. If we do not pull or build at the time of VM creation then the end user will not have the latest code in their VM. 

Hpoe this answer's your question. We can jump on a call quickly and discuss more if you need more details.